framework/src/modules/nft/method.ts

```diff
@@ -420,4 +433,196 @@ export class NFTMethod extends BaseMethod {
+		const nftExists = await nftStore.has(methodContext, nftID);
+		if (!nftExists) {
+			this.events.get(TransferEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(TransferEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+			throw new Error('NFT is escrowed to another chain');
+		}
+
+		if (!owner.equals(senderAddress)) {
+			this.events.get(TransferEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					nftID,
+				},
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+			throw new Error('Transfer not initiated by the NFT owner');
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const userData = await userStore.get(methodContext, userStore.getKey(owner, nftID));
+		if (userData.lockingModule !== NFT_NOT_LOCKED) {
+			this.events.get(TransferEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					nftID,
+				},
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+			throw new Error('Locked NFTs cannot be transferred');
+		}
+
+		await this._internalMethod.transferInternal(methodContext, recipientAddress, nftID);
+	}
+
+	public async transferCrossChain(
+		methodContext: MethodContext,
+		senderAddress: Buffer,
+		recipientAddress: Buffer,
+		nftID: Buffer,
+		receivingChainID: Buffer,
+		messageFee: bigint,
+		data: string,
+		includeAttributes: boolean,
+	): Promise<void> {
+		if (data.length > MAX_LENGTH_DATA) {
+			this.events.get(TransferCrossChainEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_DATA_TOO_LONG,
+			);
+			throw new Error('Data field is too long');
+		}
+
+		const nftStore = this.stores.get(NFTStore);
+		const nftExists = await nftStore.has(methodContext, nftID);
+		if (!nftExists) {
+			this.events.get(TransferCrossChainEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_DOES_NOT_EXIST,
+			);
+			throw new Error('NFT substore entry does not exist');
+		}
+
+		const owner = await this.getNFTOwner(methodContext, nftID);
+		if (owner.length === LENGTH_CHAIN_ID) {
+			this.events.get(TransferCrossChainEvent).error(
+				methodContext,
+				{
+					senderAddress,
+					recipientAddress,
+					receivingChainID,
+					nftID,
+					includeAttributes,
+				},
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+			throw new Error('NFT is escrowed to another chain');
+		}
+
```

Comment:
It seems to me that check if `chainID` is equal to either the `OWN_CHAIN_ID` or `receivingChainID` is missing, see [this part](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1851-L1853) of the `transferCrossChain` verification in the LIP.

Reply:
indeed. Thanks for noticing this.